### PR TITLE
add active attribute to the GFS_typesdef.meta wherever needed

### PR DESCRIPTION
**Pull request by XiaSun-Atmos**
## Description
Add the active attribute to the vars wherever needed in the GFS_typesdef.meta file.
All the vars that are allocated in the GFS_typesdef.F90 based on `if `conditions were added an extra active attribute, except for `phy_fctd`.
In GFS_typesdef.F90: 
```
    if (Model%nctp > 0 .and. Model%cscnv) then
      allocate (Tbd%phy_fctd (IM,Model%nctp))
      Tbd%phy_fctd = clear_val
    endif
```
When 
`active = (number_of_cloud_types_CS > 0 .and. flag_for_Chikira_Sugiyama_deep_convection) `
is added to `phy_fctd `in GFS_typesdef.meta, error message is `'Exception: Variable number_of_cloud_types_cs used in conditional for cloud_base_mass_flux not known to host model`

Further inspection reveals that this is related to the upper case used in the active attribute. Somehow `items = FORTRAN_CONDITIONAL_REGEX.findall(var.active.lower())` in the ` mkstatic.py` only returns lower case strings. I have tried `items = FORTRAN_CONDITIONAL_REGEX.findall(var.active)`, then `print (items)`, they are still lower case string.

Currently, the active attribute needed for `phy_fctd ` is not included yet.

## Testing
All 57 regression tests passed on Hera using the intel compiler with the current updated GFS_typesdef.meta file

## Dependencies
[ufs-community/ufs-weather-model #182](https://github.com/ufs-community/ufs-weather-model/pull/182)

### CODE REVIEW

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 1]
```diff
@@ -6004,41 +6201,47 @@
   dimensions = (horizonal_dimension,number_of_dust_bins_for_diagnostics)
   type = real
   kind = kind_phys
+  active = (number_of_chemical_tracers > 0)
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Should this be
```
active = (number_of_chemical_tracers > 0 .and. number_of_dust_bins_for_diagnostics > 0)
```
?

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 2]
```diff
@@ -6004,41 +6201,47 @@
...
 [ssem]
   standard_name = instantaneous_seasalt_emission_flux
   long_name = instantaneous sea salt emission flux
   units = kg m-2 s-1
   dimensions = (horizonal_dimension,number_of_seasalt_bins_for_diagnostics)
   type = real
   kind = kind_phys
+  active = (number_of_chemical_tracers > 0)
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Should this be
```
active = (number_of_chemical_tracers > 0 .and. number_of_seasalt_bins_for_diagnostics > 0)
```
?

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 3]
```diff
@@ -6004,41 +6201,47 @@
...
 [sedim]
   standard_name = instantaneous_sedimentation
   long_name = instantaneous sedimentation
   units = kg m-2 s-1
   dimensions = (horizonal_dimension,number_of_chemical_tracers_for_diagnostics)
   type = real
   kind = kind_phys
+  active = (index_for_first_chemical_tracer > 0 .and. number_of_chemical_tracers > 0)
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: I know the logic is confusing, but I think it suffices to say
```
active = (number_of_chemical_tracers_for_diagnostics > 0)
```

File: gfsphysics/GFS_layer/GFS_typedefs.meta (hunk 2 above)
Comment (climbfuji) [reply gfsphysics/GFS_layer/GFS_typedefs.meta]: Reading it again, I think we can just say
```
active = (number_of_seasalt_bins_for_diagnostics > 0)
```

File: gfsphysics/GFS_layer/GFS_typedefs.meta (hunk 1 above)
Comment (climbfuji) [reply gfsphysics/GFS_layer/GFS_typedefs.meta]: Reading it again, I think we can just say
```
active = (number_of_dust_bins_for_diagnostics > 0)
```

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 4]
```diff
@@ -6004,41 +6201,47 @@
...
 [drydep]
   standard_name = instantaneous_dry_deposition
   long_name = instantaneous dry deposition
   units = kg m-2 s-1
   dimensions = (horizonal_dimension,number_of_chemical_tracers_for_diagnostics)
   type = real
   kind = kind_phys
+  active = (index_for_first_chemical_tracer > 0 .and. number_of_chemical_tracers > 0)
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Same as for `sedim`.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 5]
```diff
@@ -6004,41 +6201,47 @@
...
 [wetdpl]
   standard_name = instantaneous_large_scale_wet_deposition
   long_name = instantaneous large-scale wet deposition
   units = kg m-2 s-1
   dimensions = (horizonal_dimension,number_of_chemical_tracers_for_diagnostics)
   type = real
   kind = kind_phys
+  active = (index_for_first_chemical_tracer > 0 .and. number_of_chemical_tracers > 0)
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Same as for `sedim`.

File: gfsphysics/GFS_layer/GFS_typedefs.meta [hunk 6]
```diff
@@ -6004,41 +6201,47 @@
...
 [wetdpc]
   standard_name = instantaneous_convective_scale_wet_deposition
   long_name = instantaneous convective-scale wet deposition
   units = kg m-2 s-1
   dimensions = (horizonal_dimension,number_of_chemical_tracers_for_diagnostics)
   type = real
   kind = kind_phys
+  active = (index_for_first_chemical_tracer > 0 .and. number_of_chemical_tracers > 0)
```
Comment (climbfuji) [inline gfsphysics/GFS_layer/GFS_typedefs.meta]: Same as for `sedim`.

**Review by climbfuji — CHANGES_REQUESTED** [review]
This is great work, @XiaSun-NOAA. I did a first pass through the entire file, and it looks pretty good to me. Only a few small changes, but please double check that my logic makes sense. Will need to do another pass one when it's our time to merge (there will be some updates to the variables in `GFS_typedefs.F90`, especially because of the recent RRTMGP commit).

File: gfsphysics/GFS_layer/GFS_typedefs.meta (hunk 1 above)
Comment (XiaSun-Atmos) [reply gfsphysics/GFS_layer/GFS_typedefs.meta]: Thank you @climbfuji, it makes sense. I have updated the file in the following commit.

**Comment by XiaSun-Atmos** [comment]
Thank you @climbfuji . I have made corresponding modifications and will run the RT tests when it is time.

**Comment by climbfuji** [comment]
Thanks very much for this, you'll need to pull in the updates from the authoritative repositories at that time, too. Probably look at what as changed in GFS_typedefs.F90 before/after the update from develop and see if anything needs to be changed in GFS_typedefs.meta.

**Comment by XiaSun-Atmos** [comment]
@climbfuji Hi Dom, I included the recent changes of GFS_typedef.F90 into the meta file. RT tests are waiting in the queue and might take some time to start running. Thanks!

**Comment by XiaSun-Atmos** [comment]
Recent land-surface perturbation changes are included in the GFS_typdefs.meta file.

**Comment by climbfuji** [comment]
This PR was pulled into https://github.com/NOAA-EMC/fv3atm/pull/153 and the missing `active` attribute for `phy_fctd` was added (after making the necessary changes in ccpp-framework).

**Comment by climbfuji** [comment]
@XiaSun-NOAA this was merged as part of https://github.com/NOAA-EMC/fv3atm/pull/153, please close the PR.

**Pull request closed without merging**